x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```diff
@@ -0,0 +1,482 @@
+            builder = this;
+        }
+
+        public Builder ignoreMalformed(boolean ignoreMalformed) {
+            this.ignoreMalformed = ignoreMalformed;
+            return builder;
+        }
+
+        protected Explicit<Boolean> ignoreMalformed(BuilderContext context) {
+            if (ignoreMalformed != null) {
+                return new Explicit<>(ignoreMalformed, true);
+            }
+            if (context.indexSettings() != null) {
+                return new Explicit<>(IGNORE_MALFORMED_SETTING.get(context.indexSettings()), false);
+            }
+            return HistogramFieldMapper.Defaults.IGNORE_MALFORMED;
+        }
+
+        @Override
+        public Builder store(boolean store) {
+            if (store) {
+                throw new IllegalArgumentException("The [" + CONTENT_TYPE + "] field does not support " +
+                    "stored fields");
+            }
+            return super.store(false);
+        }
+
+        @Override
+        public Builder index(boolean index) {
+            if (index) {
+                throw new IllegalArgumentException("The [" + CONTENT_TYPE + "] field does not support indexing");
+            }
+            return super.store(false);
+        }
+
+        @Override
+        public Builder indexOptions(IndexOptions indexOptions) {
+            if (indexOptions.equals(IndexOptions.NONE) == false) {
+                throw new IllegalArgumentException("The [" + CONTENT_TYPE + "] field does not support " +
+                    "index options, got [index_options]=" + indexOptionToString(indexOptions));
+            }
+            return super.indexOptions(indexOptions);
+        }
+
+        public HistogramFieldMapper build(BuilderContext context, String simpleName, MappedFieldType fieldType,
+                                          MappedFieldType defaultFieldType, Settings indexSettings,
+                                          MultiFields multiFields, Explicit<Boolean> ignoreMalformed, CopyTo copyTo) {
+            setupFieldType(context);
+            return new HistogramFieldMapper(simpleName, fieldType, defaultFieldType, indexSettings, multiFields,
+                ignoreMalformed, copyTo);
+        }
+
+        @Override
+        public HistogramFieldMapper build(BuilderContext context) {
+            return build(context, name, fieldType, defaultFieldType, context.indexSettings(),
+                multiFieldsBuilder.build(this, context), ignoreMalformed(context), copyTo);
+        }
+    }
+
+    public static class TypeParser implements Mapper.TypeParser {
+        @Override
+        public Mapper.Builder<Builder, HistogramFieldMapper> parse(String name,
+                                                                   Map<String, Object> node, ParserContext parserContext)
+                throws MapperParsingException {
+            Builder builder = new HistogramFieldMapper.Builder(name);
+            parseField(builder, name, node, parserContext);
```

Comment:
Let's not call `parseField`, none of the properties that it parses are supported by this field?

Reply:
I am supporting them:

* user can disable docValues
* If user set index, index options or stored values, an error will be thrown.

I am not handling boost, similarity and copy_to. Shall we throw an error if user defines those fields?